packages/trader/src/RangeTrader.ts

```diff
@@ -0,0 +1,183 @@
+  readonly formatDecimalString: any;
+  readonly exchangeAdapter: any;
+
+  // TODO: is there any way to not have to declare these as part of the type?
+  readonly BN: any;
+  readonly toBN: any;
+  readonly toWei: any;
+  readonly fromWei: any;
+
+  readonly tradeExecutionThreshold: any;
+  readonly targetPriceSpread: any;
+  readonly fixedPointAdjustment: any;
+
+  constructor(
+    /**
+   * @notice Constructs new Range Trader.
+   * @param {Object} logger Module used to send logs.
+   * @param {Object} web3 Provider from Truffle/node  to connect to Ethereum network.
+   * @param {Object} tokenPriceFeed Price feed to fetch the current synthetic token trading price. EG a Dex price feed.
+   * @param {Object} referencePriceFeed Price feed to fetch the "real" identifier price. EG a Cryptowatch price feed.
+   * @param {Object} exchangeAdapter Interface to interact with on-chain exchange. EG: Uniswap.
+   * @param {Object} rangeTraderConfig: Config to parameterize the range trader. Expected:
+   *      { tradeExecutionThreshold: 0.2,  -> error amount which must be exceeded for a correcting trade to be executed.
+            targetPriceSpread: 0.05 }      -> target price that should be present after a correcting trade has concluded.
+   */
+    logger: any,
+    web3: any,
+    tokenPriceFeed: any,
+    referencePriceFeed: any,
+    exchangeAdapter: any,
+    rangeTraderConfig: any
+  ) {
+    assert(tokenPriceFeed.priceFeedDecimals === referencePriceFeed.priceFeedDecimals, "pricefeed decimals must match");
+
+    this.logger = logger;
+    this.web3 = web3;
+    this.tokenPriceFeed = tokenPriceFeed;
+    this.referencePriceFeed = referencePriceFeed;
+    this.exchangeAdapter = exchangeAdapter;
+
+    this.normalizePriceFeedDecimals = ConvertDecimals(tokenPriceFeed.priceFeedDecimals, 18, this.web3);
+
+    // Formats an 18 decimal point string with a define number of decimals and precision for use in message generation.
+    this.formatDecimalString = createFormatFunction(this.web3, 2, 4, false);
+
+    // Helper functions from web3.
+    this.BN = this.web3.utils.BN;
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+    this.fromWei = this.web3.utils.fromWei;
+
+    // Default config settings.
+    const defaultConfig = {
+      tradeExecutionThreshold: {
+        value: 0.2,
+        isValid: (x: Number) => {
+          return x > 0;
+        }
+      },
+      targetPriceSpread: {
+        value: 0.05,
+        isValid: (x: Number) => {
+          return x > 0 && x <= 1;
+        }
+      }
+    };
+
+    // Validate and set config settings to class state.
+    const configWithDefaults = createObjectFromDefaultProps(rangeTraderConfig, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    this.fixedPointAdjustment = this.toBN(this.toWei("1"));
+  }
+
+  async update() {
+    await Promise.all([this.tokenPriceFeed.update(), this.referencePriceFeed.update()]);
```

Comment:
is this function needed anymore, considering it can be done as needed in other functions?

Reply:
not strictly; we could put the updates within the parent class. I've done just this. it's a bit cleaner.